lib/collection/src/shards/telemetry.rs

```diff
@@ -58,6 +58,7 @@ pub struct LocalShardTelemetry {
     /// Do NOT rely on this number unless you know what you are doing
     #[serde(skip_serializing_if = "Option::is_none")]
     pub num_vectors: Option<usize>,
+    #[serde(skip_serializing_if = "Vec::is_empty")]
```

Comment:
Empty vectors will break OpenAPI validation - it should be `Option<Vec<SegmentTelemetry>>` instead